src/ai/backend/manager/models/minilang/queryfilter.py

```diff
@@ -50,6 +51,20 @@
 FieldSpecType: TypeAlias = Mapping[str, FieldSpecItem] | None
 
 
+def enum_field_getter(enum_cls: Type[Enum]) -> Callable[[str], Enum]:
```

Comment:
Could we use generic types here? Or, is it sufficient to use just `enum.Enum`?

Reply:
I tried but `enum.Enum` does not pass type-checking and I have no idea how to use generic types here.
this is what failure I got when I run type-checking with `enum.Enum`.
<img width="1360" alt="image" src="https://github.com/lablup/backend.ai/assets/44239739/b4dc3093-e24a-4fc1-96d7-e681260263fd">